drake/systems/plants/BotVisualizer.h

```diff
@@ -11,125 +11,133 @@
+              gdata.num_float_data = 2;
+              auto c = dynamic_cast<const DrakeShapes::Cylinder &>(geometry);
+              gdata.float_data.push_back(static_cast<float>(c.radius));
+              gdata.float_data.push_back(static_cast<float>(c.length));
+              break;
+            }
+            case DrakeShapes::MESH: {
+              gdata.type = gdata.MESH;
+              gdata.num_float_data = 1;
+              auto m = dynamic_cast<const DrakeShapes::Mesh &>(geometry);
+              gdata.float_data.push_back(static_cast<float>(m.scale));
+              gdata.string_data = m.resolved_filename;  // looks like this could be empty, but it is what's used in the get mesh points...
+              break;
+            }
+            case DrakeShapes::CAPSULE: {
+              gdata.type = gdata.CAPSULE;
+              gdata.num_float_data = 2;
+              auto c = dynamic_cast<const DrakeShapes::Capsule &>(geometry);
+              gdata.float_data.push_back(static_cast<float>(c.radius));
+              gdata.float_data.push_back(static_cast<float>(c.length));
+              break;
+            }
+            default: {
+              // intentionally do nothing
+              break;
+            }
           }
-        }
 
-        Eigen::Matrix4d T = v.getLocalTransform();
-        Eigen::Map<Eigen::Vector3f> position(gdata.position);
-        position = T.topRightCorner<3, 1>().cast<float>();
-        Eigen::Map<Eigen::Vector4f> quaternion(gdata.quaternion);
-        quaternion = rotmat2quat(T.topLeftCorner<3, 3>()).cast<float>();
+          Eigen::Matrix4d T = v.getLocalTransform();
+          Eigen::Map<Eigen::Vector3f> position(gdata.position);
+          position = T.topRightCorner<3, 1>().cast<float>();
+          Eigen::Map<Eigen::Vector4f> quaternion(gdata.quaternion);
+          quaternion = rotmat2quat(T.topLeftCorner<3, 3>()).cast<float>();
 
-        Eigen::Map<Eigen::Vector4f> color(gdata.color);
-        color = v.getMaterial().cast<float>();
+          Eigen::Map<Eigen::Vector4f> color(gdata.color);
+          color = v.getMaterial().template cast<float>();
 
-        link.geom.push_back(gdata);
+          link.geom.push_back(gdata);
+        }
+        vr.link.push_back(link);
       }
-      vr.link.push_back(link);
-    }
 
-    lcm->publish("DRAKE_VIEWER_LOAD_ROBOT", &vr);
-  }
+      lcm->publish("DRAKE_VIEWER_LOAD_ROBOT", &vr);
+    }
 
-  virtual Eigen::VectorXd output(double t, const Eigen::VectorXd& unused, const Eigen::VectorXd& u) const override {
-    draw_msg.timestamp = static_cast<int64_t>(t*1000.0);
+    UnusedVector<double> outputImplementation(const double& t, const InputVector<double> &u) const {
+      draw_msg.timestamp = static_cast<int64_t>(t * 1000.0);
+
+      Eigen::Matrix<double,VectorTraits<InputVector<double> >::RowsAtCompileTime,1> uvec(u);
+      auto q = uvec.head(manip.num_positions);
+      KinematicsCache<double> cache = manip.doKinematics(q);
```

Comment:
Would be a lot more efficient to create a KinematicsCache as a member variable and reuse it.